File: app/src/main/java/com/zy/mei_kotlin/base/BaseViewActivity.java
package com.zy.mei_kotlin.base;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

/**
 * Created by Simon on 2017/5/25.
 */

public abstract class BaseViewActivity<T extends BaseActivityPresenter> extends BaseActivity implements BaseView<T> {


    public static final class ViewActivityLifecycle implements Application.ActivityLifecycleCallbacks {


        @Override
        public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
            if (!(activity instanceof BaseViewActivity)) {
                return;
            }
            if (((BaseViewActivity) activity).getPresenter() == null) {
                return;
            }
            ((BaseViewActivity) activity).getPresenter().onActivityCreate(savedInstanceState);
        }

        @Override
        public void onActivityStarted(Activity activity) {
            if (!(activity instanceof BaseViewActivity)) {
                return;
            }
            if (((BaseViewActivity) activity).getPresenter() == null) {
                return;
            }
            ((BaseViewActivity) activity).getPresenter().onActivityStarted();
        }

        @Override
        public void onActivityResumed(Activity activity) {
            if (!(activity instanceof BaseViewActivity)) {
                return;
            }
            if (((BaseViewActivity) activity).getPresenter() == null) {
                return;
            }
            ((BaseViewActivity) activity).getPresenter().onActivityResumed();
        }

        @Override
        public void onActivityPaused(Activity activity) {
            if (!(activity instanceof BaseViewActivity)) {
                return;
            }
            if (((BaseViewActivity) activity).getPresenter() == null) {
                return;
            }
            ((BaseViewActivity) activity).getPresenter().onActivityPaused();
        }

        @Override
        public void onActivityStopped(Activity activity) {
            if (!(activity instanceof BaseViewActivity)) {
                return;
            }
            if (((BaseViewActivity) activity).getPresenter() == null) {
                return;
            }
            ((BaseViewActivity) activity).getPresenter().onActivityStopped();
        }

        @Override
        public void onActivitySaveInstanceState(Activity activity, Bundle outState) {
            if (!(activity instanceof BaseViewActivity)) {
                return;
            }
            if (((BaseViewActivity) activity).getPresenter() == null) {
                return;
            }
            ((BaseViewActivity) activity).getPresenter().onActivitySaveInstanceState(outState);
        }

        @Override
        public void onActivityDestroyed(Activity activity) {
            if (!(activity instanceof BaseViewActivity)) {
                return;
            }
            if (((BaseViewActivity) activity).getPresenter() == null) {
                return;
            }
            ((BaseViewActivity) activity).getPresenter().onActivityDestroyed();
        }
    }
}

File: app/src/main/java/com/zy/mei_kotlin/base/BaseViewActivity.kt
package com.zy.mei_kotlin.base

import android.app.Activity
import android.app.Application
import android.os.Bundle

/**
 * Created by Simon on 2017/5/22.
 */
abstract class BaseViewActivity<T : BasePresent> : BaseActivity(), BaseView<T> {


    class ViewLifecycle : Application.ActivityLifecycleCallbacks {

        override fun onActivityCreated(p0: Activity?, p1: Bundle?) {
            if (p0 !is BaseView<*>) {
                return
            }

            p0.present?.onCreate(p1)
        }

        override fun onActivityStarted(p0: Activity?) {
            if (p0 is BaseView<*>) {
                p0.present?.onStarted()
            }
        }

        override fun onActivityResumed(p0: Activity?) {
            if (p0 is BaseView<*>) {
                p0.present?.onResumed()
            }
        }


        override fun onActivityPaused(p0: Activity?) {
            if (p0 is BaseView<*>) {
                p0.present?.onPaused()
            }
        }

        override fun onActivityStopped(p0: Activity?) {
            if (p0 is BaseView<*>) {
                p0.present?.onStopped()
            }
        }

        override fun onActivityDestroyed(p0: Activity?) {
            if (p0 is BaseView<*>) {
                p0.present?.onDestroyed()
            }
        }

        override fun onActivitySaveInstanceState(p0: Activity?, p1: Bundle?) {
            if (p0 is BaseView<*>) {
                p0.present?.onSaveInstanceState(p1)
            }
        }

    }
}